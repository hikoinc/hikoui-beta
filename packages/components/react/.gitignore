# Dependencies
node_modules

# Build outputs
dist

# Turbo
.turbo

# OS
.DS_Store

# Debug
*storybook.log
