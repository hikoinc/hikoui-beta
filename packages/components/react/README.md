# @hikoui/react

HikoUI is a minimal-design React + Tailwind CSS component system for fast, scalable UI development.

> Build clean, simple, and consistent UIs with minimal design — quickly and efficiently.

## Contributing

We’re building HikoUI openly and plan to welcome community contributions soon.

For now, feel free to explore the code and see our [contributing guide](../../CONTRIBUTING.md).

## License

MIT License — see [LICENSE](../../LICENSE).
