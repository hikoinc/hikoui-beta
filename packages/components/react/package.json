{
  "name": "@hikoui-beta/react",
  "version": "0.0.2",
  "description": "A minimalist React component library built with Tailwind CSS for fast, scalable UI development.",
  "keywords": [
    "react",
    "ui",
    "components",
    "tailwindcss",
    "typescript",
    "design",
    "minimal",
    "frontend"
  ],
  "homepage": "https://hikoui.com",
  "bugs": {
    "url": "https://github.com/hikoinc/hikoui/issues"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/hikoinc/hikoui.git",
    "directory": "packages/component/react"
  },
  "license": "MIT",
  "author": "Joe Takara <https://github.com/joetakara>",
  "sideEffects": false,
  "exports": {
    ".": {
      "types": "./dist/index.d.ts",
      "import": "./dist/index.mjs",
      "require": "./dist/index.js"
    },
    "./package.json": "./package.json"
  },
  "main": "dist/index.js",
  "module": "dist/index.mjs",
  "types": "dist/index.d.ts",
  "files": [
    "dist"
  ],
  "scripts": {
    "build": "pnpm clean:build && tsup",
    "clean": "pnpm clean:build && pnpm clean:turbo && pnpm clean:node_modules",
    "clean:build": "rimraf dist",
    "clean:node_modules": "rimraf node_modules",
    "clean:turbo": "rimraf .turbo",
    "dev": "pnpm clean:build && tsup --watch",
    "sort-package-json": "pnpm exec sort-package-json",
    "sort-package-json:check": "pnpm exec sort-package-json --check",
    "typescheck": "tsc --noEmit"
  },
  "dependencies": {
    "@hikoui-beta/system": "workspace:*",
    "@react-types/shared": "^3.29.1",
    "class-variance-authority": "^0.7.1",
    "html-element-attributes": "^3.4.0",
    "lodash": "^4.17.21",
    "react": "19.1.0",
    "react-dom": "19.1.0",
    "tailwind-merge": "^3.3.0"
  },
  "devDependencies": {
    "@tailwindcss/postcss": "^4.1.8",
    "@testing-library/jest-dom": "^6.6.3",
    "@testing-library/react": "^16.3.0",
    "@types/lodash": "^4.17.17",
    "@types/react": "19.1.0",
    "@types/react-dom": "19.1.1",
    "postcss": "^8.5.4",
    "sort-package-json": "^3.0.0",
    "tailwindcss": "^4.1.8",
    "tsup": "^8.5.0",
    "typescript": "5.8.2"
  },
  "peerDependencies": {
    "react": ">=19",
    "react-dom": ">=19"
  },
  "packageManager": "pnpm@10.11.0",
  "engines": {
    "node": ">=20"
  },
  "publishConfig": {
    "access": "public"
  }
}
