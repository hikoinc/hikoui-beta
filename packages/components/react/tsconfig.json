{
  "extends": "../../../tsconfig.json",
  "compilerOptions": {
    "outDir": "dist",
    "jsx": "react-jsx"
  },
  "include": ["src", "turbo"],
  "exclude": ["node_modules", "dist"]
}
