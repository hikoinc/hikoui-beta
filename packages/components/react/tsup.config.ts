import { defineConfig } from "tsup";

export default defineConfig({
  entry: ["src/index.ts"],
  format: ["cjs", "esm"],
  clean: true,
  target: "es2022",
  watch: true,
  dts: true,
  banner: {
    js: '"use client";',
  },
});
