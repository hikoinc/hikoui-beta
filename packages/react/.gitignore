# Dependencies
node_modules

# Build outputs
dist

# Debug
.pnpm-debug.log*

# OS
.DS_Store

# Turbo
.turbo
