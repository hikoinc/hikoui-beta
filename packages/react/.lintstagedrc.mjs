/* eslint-env node */

import { relative } from "path";

import { ESLint } from "eslint";

const removeIgnoredFiles = async (files) => {
  const cwd = process.cwd();
  const eslint = new ESLint();
  const relativePaths = files.map((file) => relative(cwd, file));
  const isIgnored = await Promise.all(relativePaths.map((file) => eslint.isPathIgnored(file)));
  const filteredFiles = files.filter((_, i) => !isIgnored[i]);

  return filteredFiles.join(" ");
};

export default {
  "**/*.{js,ts,jsx,tsx}": async (files) => {
    const filesToLint = await removeIgnoredFiles(files);

    return [`eslint -c .eslintrc.json --max-warnings=0 --fix ${filesToLint}`];
  },
  "**/*.css": async (files) => {
    const filesToLint = await removeIgnoredFiles(files);

    return [`prettier --config .prettierrc.json --ignore-path --write ${filesToLint}`];
  },
};
