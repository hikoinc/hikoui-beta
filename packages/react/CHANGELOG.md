# @hikoui-beta/react

## 0.0.1

### Patch Changes

- [#12](https://github.com/hikoinc/hikoui-beta/pull/12) [`948d583`](https://github.com/hikoinc/hikoui-beta/commit/948d5839fb5c14c0c398f501ccfb5baed3156973) Thanks [@joetakara](https://github.com/joetakara)! - Test
