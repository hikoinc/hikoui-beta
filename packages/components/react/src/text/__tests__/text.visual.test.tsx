import { render } from "@testing-library/react";

import "@testing-library/jest-dom";
import Text from "../text";

// Visual regression tests using Jest snapshots
// These tests capture the rendered HTML structure and CSS classes
// to detect unintended visual changes
describe("Text - Visual Regression Tests", () => {
  describe("basic variants", () => {
    it("matches snapshot for default text", () => {
      const { container } = render(<Text>Default text</Text>);
      expect(container.firstChild).toMatchSnapshot();
    });

    it("matches snapshot for all color variants", () => {
      const colors = [
        "primary",
        "secondary",
        "tertiary",
        "success",
        "warning",
        "danger",
        "info",
        "muted",
        "inherit",
      ] as const;

      colors.forEach((color) => {
        const { container } = render(<Text color={color}>Text with {color} color</Text>);
        expect(container.firstChild).toMatchSnapshot(`text-color-${color}`);
      });
    });

    it("matches snapshot for all size variants", () => {
      const sizes = [
        "2xs",
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
        "2xl",
        "3xl",
        "4xl",
        "5xl",
        "6xl",
        "7xl",
        "8xl",
        "9xl",
      ] as const;

      sizes.forEach((size) => {
        const { container } = render(<Text size={size}>Text with {size} size</Text>);
        expect(container.firstChild).toMatchSnapshot(`text-size-${size}`);
      });
    });

    it("matches snapshot for all weight variants", () => {
      const weights = [
        "thin",
        "extralight",
        "light",
        "normal",
        "medium",
        "semibold",
        "bold",
        "extrabold",
        "black",
      ] as const;

      weights.forEach((weight) => {
        const { container } = render(<Text weight={weight}>Text with {weight} weight</Text>);
        expect(container.firstChild).toMatchSnapshot(`text-weight-${weight}`);
      });
    });
  });

  describe("combined variants", () => {
    it("matches snapshot for heading style", () => {
      const { container } = render(
        <Text as="h1" size="3xl" weight="bold" color="primary">
          Main Heading
        </Text>,
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it("matches snapshot for subheading style", () => {
      const { container } = render(
        <Text as="h2" size="xl" weight="semibold" color="secondary">
          Subheading
        </Text>,
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it("matches snapshot for body text style", () => {
      const { container } = render(
        <Text size="md" color="primary" leading="relaxed">
          This is a body text with relaxed leading for better readability.
        </Text>,
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it("matches snapshot for caption style", () => {
      const { container } = render(
        <Text size="sm" color="muted" fontStyle="italic">
          Image caption or small text
        </Text>,
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it("matches snapshot for code style", () => {
      const { container } = render(
        <Text as="code" size="sm" className="rounded bg-gray-100 px-2 py-1">
          const example = `code`;
        </Text>,
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe("special styling", () => {
    it("matches snapshot for truncated text", () => {
      const { container } = render(
        <Text lineClamp="2" className="w-64">
          This is a very long text that should be truncated at exactly two lines when displayed in a container with
          limited width.
        </Text>,
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it("matches snapshot for decorated text", () => {
      const { container } = render(
        <Text decoration="underline" transform="uppercase" letterSpacing="wide">
          Decorated Text
        </Text>,
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it("matches snapshot for aligned text", () => {
      const alignments = ["left", "center", "right", "justify"] as const;

      alignments.forEach((align) => {
        const { container } = render(
          <Text align={align} className="w-64">
            Text with {align} alignment in a fixed width container
          </Text>,
        );
        expect(container.firstChild).toMatchSnapshot(`text-align-${align}`);
      });
    });
  });

  describe("accessibility states", () => {
    it("matches snapshot for error state", () => {
      const { container } = render(
        <Text color="danger" role="alert" aria-live="polite">
          Error: Something went wrong
        </Text>,
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it("matches snapshot for success state", () => {
      const { container } = render(
        <Text color="success" role="status" aria-live="polite">
          Success: Operation completed
        </Text>,
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it("matches snapshot for loading state", () => {
      const { container } = render(
        <Text color="muted" aria-live="polite" aria-busy="true">
          Loading...
        </Text>,
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe("custom styling integration", () => {
    it("matches snapshot with custom classes", () => {
      const { container } = render(
        <Text className="custom-shadow transition-colors hover:text-blue-500" size="lg">
          Text with custom styling
        </Text>,
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it("matches snapshot for dark mode preparation", () => {
      const { container } = render(
        <Text className="dark:bg-gray-800 dark:text-white" color="inherit">
          Dark mode compatible text
        </Text>,
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
