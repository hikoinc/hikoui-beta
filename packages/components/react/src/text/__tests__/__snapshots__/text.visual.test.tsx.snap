// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text - Visual Regression Tests accessibility states matches snapshot for error state 1`] = `
<span
  aria-live="polite"
  class="antialiased text-danger-text text-base"
>
  Error: Something went wrong
</span>
`;

exports[`Text - Visual Regression Tests accessibility states matches snapshot for loading state 1`] = `
<span
  aria-busy="true"
  aria-live="polite"
  class="antialiased text-muted-text text-base"
>
  Loading...
</span>
`;

exports[`Text - Visual Regression Tests accessibility states matches snapshot for success state 1`] = `
<span
  aria-live="polite"
  class="antialiased text-success-text text-base"
>
  Success: Operation completed
</span>
`;

exports[`Text - Visual Regression Tests basic variants matches snapshot for all color variants: text-color-danger 1`] = `
<span
  class="antialiased text-danger-text text-base"
>
  Text with 
  danger
   color
</span>
`;

exports[`Text - Visual Regression Tests basic variants matches snapshot for all color variants: text-color-info 1`] = `
<span
  class="antialiased text-info-text text-base"
>
  Text with 
  info
   color
</span>
`;

exports[`Text - Visual Regression Tests basic variants matches snapshot for all color variants: text-color-inherit 1`] = `
<span
  class="antialiased text-inherit text-base"
>
  Text with 
  inherit
   color
</span>
`;

exports[`Text - Visual Regression Tests basic variants matches snapshot for all color variants: text-color-muted 1`] = `
<span
  class="antialiased text-muted-text text-base"
>
  Text with 
  muted
   color
</span>
`;

exports[`Text - Visual Regression Tests basic variants matches snapshot for all color variants: text-color-primary 1`] = `
<span
  class="antialiased text-primary-text text-base"
>
  Text with 
  primary
   color
</span>
`;

exports[`Text - Visual Regression Tests basic variants matches snapshot for all color variants: text-color-secondary 1`] = `
<span
  class="antialiased text-secondary-text text-base"
>
  Text with 
  secondary
   color
</span>
`;

exports[`Text - Visual Regression Tests basic variants matches snapshot for all color variants: text-color-success 1`] = `
<span
  class="antialiased text-success-text text-base"
>
  Text with 
  success
   color
</span>
`;

exports[`Text - Visual Regression Tests basic variants matches snapshot for all color variants: text-color-tertiary 1`] = `
<span
  class="antialiased text-tertiary-text text-base"
>
  Text with 
  tertiary
   color
</span>
`;

exports[`Text - Visual Regression Tests basic variants matches snapshot for all color variants: text-color-warning 1`] = `
<span
  class="antialiased text-warning-text text-base"
>
  Text with 
  warning
   color
</span>
`;

exports[`Text - Visual Regression Tests basic variants matches snapshot for all size variants: text-size-2xl 1`] = `
<span
  class="antialiased text-primary-text text-2xl"
>
  Text with 
  2xl
   size
</span>
`;

exports[`Text - Visual Regression Tests basic variants matches snapshot for all size variants: text-size-2xs 1`] = `
<span
  class="antialiased text-primary-text text-2xs"
>
  Text with 
  2xs
   size
</span>
`;

exports[`Text - Visual Regression Tests basic variants matches snapshot for all size variants: text-size-3xl 1`] = `
<span
  class="antialiased text-primary-text text-3xl"
>
  Text with 
  3xl
   size
</span>
`;

exports[`Text - Visual Regression Tests basic variants matches snapshot for all size variants: text-size-4xl 1`] = `
<span
  class="antialiased text-primary-text text-4xl"
>
  Text with 
  4xl
   size
</span>
`;

exports[`Text - Visual Regression Tests basic variants matches snapshot for all size variants: text-size-5xl 1`] = `
<span
  class="antialiased text-primary-text text-5xl"
>
  Text with 
  5xl
   size
</span>
`;

exports[`Text - Visual Regression Tests basic variants matches snapshot for all size variants: text-size-6xl 1`] = `
<span
  class="antialiased text-primary-text text-6xl"
>
  Text with 
  6xl
   size
</span>
`;

exports[`Text - Visual Regression Tests basic variants matches snapshot for all size variants: text-size-7xl 1`] = `
<span
  class="antialiased text-primary-text text-7xl"
>
  Text with 
  7xl
   size
</span>
`;

exports[`Text - Visual Regression Tests basic variants matches snapshot for all size variants: text-size-8xl 1`] = `
<span
  class="antialiased text-primary-text text-8xl"
>
  Text with 
  8xl
   size
</span>
`;

exports[`Text - Visual Regression Tests basic variants matches snapshot for all size variants: text-size-9xl 1`] = `
<span
  class="antialiased text-primary-text text-9xl"
>
  Text with 
  9xl
   size
</span>
`;

exports[`Text - Visual Regression Tests basic variants matches snapshot for all size variants: text-size-lg 1`] = `
<span
  class="antialiased text-primary-text text-lg"
>
  Text with 
  lg
   size
</span>
`;

exports[`Text - Visual Regression Tests basic variants matches snapshot for all size variants: text-size-md 1`] = `
<span
  class="antialiased text-primary-text text-base"
>
  Text with 
  md
   size
</span>
`;

exports[`Text - Visual Regression Tests basic variants matches snapshot for all size variants: text-size-sm 1`] = `
<span
  class="antialiased text-primary-text text-sm"
>
  Text with 
  sm
   size
</span>
`;

exports[`Text - Visual Regression Tests basic variants matches snapshot for all size variants: text-size-xl 1`] = `
<span
  class="antialiased text-primary-text text-xl"
>
  Text with 
  xl
   size
</span>
`;

exports[`Text - Visual Regression Tests basic variants matches snapshot for all size variants: text-size-xs 1`] = `
<span
  class="antialiased text-primary-text text-xs"
>
  Text with 
  xs
   size
</span>
`;

exports[`Text - Visual Regression Tests basic variants matches snapshot for all weight variants: text-weight-black 1`] = `
<span
  class="antialiased text-primary-text text-base font-black"
>
  Text with 
  black
   weight
</span>
`;

exports[`Text - Visual Regression Tests basic variants matches snapshot for all weight variants: text-weight-bold 1`] = `
<span
  class="antialiased text-primary-text text-base font-bold"
>
  Text with 
  bold
   weight
</span>
`;

exports[`Text - Visual Regression Tests basic variants matches snapshot for all weight variants: text-weight-extrabold 1`] = `
<span
  class="antialiased text-primary-text text-base font-extrabold"
>
  Text with 
  extrabold
   weight
</span>
`;

exports[`Text - Visual Regression Tests basic variants matches snapshot for all weight variants: text-weight-extralight 1`] = `
<span
  class="antialiased text-primary-text text-base font-extralight"
>
  Text with 
  extralight
   weight
</span>
`;

exports[`Text - Visual Regression Tests basic variants matches snapshot for all weight variants: text-weight-light 1`] = `
<span
  class="antialiased text-primary-text text-base font-light"
>
  Text with 
  light
   weight
</span>
`;

exports[`Text - Visual Regression Tests basic variants matches snapshot for all weight variants: text-weight-medium 1`] = `
<span
  class="antialiased text-primary-text text-base font-medium"
>
  Text with 
  medium
   weight
</span>
`;

exports[`Text - Visual Regression Tests basic variants matches snapshot for all weight variants: text-weight-normal 1`] = `
<span
  class="antialiased text-primary-text text-base font-normal"
>
  Text with 
  normal
   weight
</span>
`;

exports[`Text - Visual Regression Tests basic variants matches snapshot for all weight variants: text-weight-semibold 1`] = `
<span
  class="antialiased text-primary-text text-base font-semibold"
>
  Text with 
  semibold
   weight
</span>
`;

exports[`Text - Visual Regression Tests basic variants matches snapshot for all weight variants: text-weight-thin 1`] = `
<span
  class="antialiased text-primary-text text-base font-thin"
>
  Text with 
  thin
   weight
</span>
`;

exports[`Text - Visual Regression Tests basic variants matches snapshot for default text 1`] = `
<span
  class="antialiased text-primary-text text-base"
>
  Default text
</span>
`;

exports[`Text - Visual Regression Tests combined variants matches snapshot for body text style 1`] = `
<span
  class="antialiased text-primary-text text-base leading-relaxed"
>
  This is a body text with relaxed leading for better readability.
</span>
`;

exports[`Text - Visual Regression Tests combined variants matches snapshot for caption style 1`] = `
<span
  class="antialiased text-muted-text text-sm italic"
>
  Image caption or small text
</span>
`;

exports[`Text - Visual Regression Tests combined variants matches snapshot for code style 1`] = `
<code
  class="antialiased text-primary-text text-sm rounded bg-gray-100 px-2 py-1"
>
  const example = "code";
</code>
`;

exports[`Text - Visual Regression Tests combined variants matches snapshot for heading style 1`] = `
<h1
  class="antialiased text-primary-text text-3xl font-bold"
>
  Main Heading
</h1>
`;

exports[`Text - Visual Regression Tests combined variants matches snapshot for subheading style 1`] = `
<h2
  class="antialiased text-secondary-text text-xl font-semibold"
>
  Subheading
</h2>
`;

exports[`Text - Visual Regression Tests custom styling integration matches snapshot for dark mode preparation 1`] = `
<span
  class="antialiased text-inherit text-base dark:bg-gray-800 dark:text-white"
>
  Dark mode compatible text
</span>
`;

exports[`Text - Visual Regression Tests custom styling integration matches snapshot with custom classes 1`] = `
<span
  class="antialiased text-primary-text text-lg custom-shadow transition-colors hover:text-blue-500"
>
  Text with custom styling
</span>
`;

exports[`Text - Visual Regression Tests special styling matches snapshot for aligned text: text-align-center 1`] = `
<span
  class="antialiased text-primary-text text-base text-center w-64"
>
  Text with 
  center
   alignment in a fixed width container
</span>
`;

exports[`Text - Visual Regression Tests special styling matches snapshot for aligned text: text-align-justify 1`] = `
<span
  class="antialiased text-primary-text text-base text-justify w-64"
>
  Text with 
  justify
   alignment in a fixed width container
</span>
`;

exports[`Text - Visual Regression Tests special styling matches snapshot for aligned text: text-align-left 1`] = `
<span
  class="antialiased text-primary-text text-base text-left w-64"
>
  Text with 
  left
   alignment in a fixed width container
</span>
`;

exports[`Text - Visual Regression Tests special styling matches snapshot for aligned text: text-align-right 1`] = `
<span
  class="antialiased text-primary-text text-base text-right w-64"
>
  Text with 
  right
   alignment in a fixed width container
</span>
`;

exports[`Text - Visual Regression Tests special styling matches snapshot for decorated text 1`] = `
<span
  class="antialiased text-primary-text text-base tracking-wide underline uppercase"
>
  Decorated Text
</span>
`;

exports[`Text - Visual Regression Tests special styling matches snapshot for truncated text 1`] = `
<span
  class="antialiased text-primary-text text-base line-clamp-2 w-64"
>
  This is a very long text that should be truncated at exactly two lines when displayed in a container with limited width.
</span>
`;
